internal/types/compare.go

```diff
@@ -27,40 +27,37 @@ import (
 )
 
 //go:generate ../../bin/stringer -linecomment -type CompareResult
+//go:generate ../../bin/stringer -linecomment -type dataTypeOrderResult
+//go:generate ../../bin/stringer -linecomment -type numberDataTypeOrderResult
+//go:generate ../../bin/stringer -linecomment -type SortType
 
 // CompareResult represents the result of a comparison.
 type CompareResult int8
 
 // Values match results of comparison functions such as bytes.Compare.
 // They do not match MongoDB `sort` values where 1 means ascending order and -1 means descending.
 const (
-	Equal   CompareResult = 0  // ==
-	Less    CompareResult = -1 // <
-	Greater CompareResult = 1  // >
-
-	// TODO Remove once it is no longer needed.
-	// It should not be needed.
-	NotEqual CompareResult = 127 // !=
+	Equal        CompareResult = 0   // ==
+	Less         CompareResult = -1  // <
+	Greater      CompareResult = 1   // >
+	Incomparable CompareResult = 127 // ≹
 )
 
-// Compare compares any BSON values in the same way as MongoDB does it for filtering or sorting.
+// Compare compares any BSON values in the same way as MongoDB does it for filtering.
 //
 // It converts types as needed; that may result in different types being equal.
 // For that reason, it typically should not be used in tests.
 //
 // Compare and contrast with test helpers in testutil package.
 func Compare(v1, v2 any) CompareResult {
-	if v1 == nil {
```

Comment:
why this change?  as for me previous version was more readable.

Reply:
returned to its original form